src/main/java/core/basesyntax/Car.java

```diff
@@ -1,13 +1,95 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 
 /**
  * Make this class immutable. See requirements in the README file
  */
-public class Car {
-    private int year;
-    private String color;
-    private List<Wheel> wheels;
-    private Engine engine;
+public final class Car {
+    private final int year;
+    private final String color;
+    private final List<Wheel> wheels;
+    private final Engine engine;
+
+    public Car(int year, String color, List<Wheel> wheels, Engine engine) {
+        if (wheels == null) {
+            throw new NullPointerException();
+        }
+        List<Wheel> newWheelsArray = new ArrayList<>();
+        for (Wheel w : wheels) {
+            newWheelsArray.add(w.clone());
+        }
+        this.wheels = newWheelsArray;
+        this.year = year;
+        this.color = color;
+        if (engine == null) {
+            this.engine = null;
+        } else {
+            this.engine = engine.clone();
+        }
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof Car)) {
+            return false;
+        }
+        Car car = (Car) o;
+        return getYear() == car.getYear()
+                && Objects.equals(getColor(), car.getColor())
+                && getWheels().equals(car.getWheels())
```

Comment:
```suggestion
                && Objects.equals(wheels, car.wheels)
```
why are you using getters?